src/enzo/DetermineSEDParameters.C

```diff
@@ -0,0 +1,290 @@
+{
+
+  if((SmartStarBHRadiativeFeedback == FALSE) && (SmartStarStellarRadiativeFeedback == FALSE)) {
+      for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+	SS->RadiationEnergyBins[bin] = 0.0;
+	SS->RadiationSED[bin] = 0.0;
+      }
+      return SUCCESS;
+    }
+  float DensityUnits, LengthUnits, TemperatureUnits, TimeUnits,
+    VelocityUnits;
+  double MassUnits;
+  GetUnits(&DensityUnits, &LengthUnits, &TemperatureUnits,
+	   &TimeUnits, &VelocityUnits, Time);
+  MassUnits = DensityUnits * POW(LengthUnits,3);
+  double MassConversion = (float) (dx*dx*dx * double(MassUnits));  //convert to g from a density
+  double ParticleMass =  SS->ReturnMass()*MassConversion/SolarMass; //In solar masses
+
+#if STELLAR
+  static float ptime=0.0;
+  if(SS->ParticleClass == POPIII || SS->ParticleClass == SMS) {
+    SS->RadiationLifetime = 1e6*3.154e7/TimeUnits; //Code Time
+    SS->LuminosityPerSolarMass = 6.696798e49/40.0; //In physical units
+
+    float smdot, dt_enzo, slum, rad;
+    smdot=SS->AccretionRate[SS->TimeIndex]*MassUnits/TimeUnits;
+    dt_enzo=Time-ptime;
+
+  
+
+    if(ptime>0.0){
+      FORTRAN_NAME(stellar)(&smdot, &dt_enzo, &slum, &rad);
+      SS->LuminosityPerSolarMass = slum/ParticleMass;
+    }
+    else{
+      SS->LuminosityPerSolarMass = 1e10;
+    }
+  }
+  ptime = Time;
+  return SUCCESS;
+#endif
+  /* 
+   * The PopIII values are taken from Schaerer et al. 2002 Table 4.
+   * Luminosity is NOT in ergs/s but in photons/s 
+   */
+  if(SS->ParticleClass == POPIII) {
+    SS->RadiationLifetime = SmartStarSMSLifetime*3.154e7/TimeUnits; //Code Time
+    SS->LuminosityPerSolarMass = 6.696798e49/40.0; //In physical units
+    /* Ideally we call SLUG here. Hardcoded for now */
+    for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+      SS->RadiationEnergyBins[bin] = PopIIIEnergyBins[bin];
+      SS->RadiationSED[bin] = PopIIISEDFracBins[bin];
+    }
+  }
+  /*
+   * For the SMS spectrum I assume a blackbody with an effective 
+   * temperature of 6000K. 
+   * Luminosity is NOT in ergs/s but in photons/s 
+   */
+  else if (SS->ParticleClass == SMS) {
+    SS->RadiationLifetime = SmartStarSMSLifetime*3.154e7/TimeUnits; //Code Times
+    SS->LuminosityPerSolarMass = 1.4e51/500.0;
+    /* Ideally we call SLUG here. Hardcoded for now */
+    for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+      SS->RadiationEnergyBins[bin] = SMSEnergyBins[bin];
+      SS->RadiationSED[bin] = SMSSEDFracBins[bin];
+    }
+  }
+  
+  /*
+   * The BH luminosity depends on the Black Hole mass and the accretion rate
+   * The values are calculated assuming a multi-colour disc for the 
+   * for accretion disk and a corona fit by a power law.
+   * The values are hardcoded into the table above and valid for black
+   * hole masses from 1e0 to 1e9 and accretion rates from 1e-7 to 1e2
+   */
+  else if(SS->ParticleClass == BH && SmartStarBHRadiativeFeedback == TRUE) {
+    SS->RadiationLifetime = 1e14*3.154e7/TimeUnits; //code time
```

Comment:
Year in seconds is hard coded.  There are multiple instances in this file where it has been done. Please, define yr_s in phys_constants.h and use here.

Reply:
Fixed